hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java

```diff
@@ -72,16 +86,30 @@
 public class ConsensusUpdateTopicHandler implements TransactionHandler {
     private static final Logger log = LogManager.getLogger(ConsensusUpdateTopicHandler.class);
 
+    private final ConsensusCustomFeesValidator customFeesValidator;
+
     /**
      * Default constructor for injection.
+     * @param customFeesValidator custom fees validator
      */
     @Inject
-    public ConsensusUpdateTopicHandler() {}
+    public ConsensusUpdateTopicHandler(@NonNull final ConsensusCustomFeesValidator customFeesValidator) {
+        requireNonNull(customFeesValidator);
+        this.customFeesValidator = customFeesValidator;
+    }
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
         final ConsensusUpdateTopicTransactionBody op = txn.consensusUpdateTopicOrThrow();
         validateTruePreCheck(op.hasTopicID(), INVALID_TOPIC_ID);
+
+        if (op.feeExemptKeyList() != null) {
```

Comment:
Could be `op.hasFeeExemptKeyList`

Reply:
Done. Even though Intelij doesn't like it and throws warning that NPE may occur.